-----------------
Strange behaviour of stock gcc of FreeBSD 7.

# gcc -print-file-name=libgcc_s.so
/usr/lib/libgcc_s.so
# gcc -print-file-name=libgcc_s.so.1
libgcc_s.so.1
. This sould be
# gcc -print-file-name=libgcc_s.so.1
/lib/libgcc_s.so.1

However gcc42 from ports behaves correctly;
and
# gcc42 -print-file-name=libgcc_s.so.1
/usr/local/lib/gcc-4.2.3/gcc/i386-portbld-freebsd7.0/4.2.3/../../../libgcc_s.so.1
# gcc42 -print-file-name=libgcc_s.so
/usr/local/lib/gcc-4.2.3/gcc/i386-portbld-freebsd7.0/4.2.3/../../../libgcc_s.so
.

Index: work/OOH680_m1/external/gcc3_specific/makefile.mk
===================================================================
RCS file: /cvs/util/external/gcc3_specific/makefile.mk,v
retrieving revision 1.5
diff -u -r1.5 makefile.mk
--- external/gcc3_specific/makefile.mk	30 Aug 2006 12:39:00 -0000	1.5
+++ external/gcc3_specific/makefile.mk	6 Jan 2008 08:10:19 -0000
@@ -12,7 +12,6 @@
 
 .IF "$(OS)"!="MACOSX"
 .IF "$(OS)"!="IRIX"
-.IF "$(OS)"!="FREEBSD"
 
 .EXPORT : CC
 
@@ -25,7 +24,6 @@
 $(LB)$/libgcc_s.so.1 :
        $(GCCINSTLIB) libgcc_s.so.1 $(LB)
 
-.ENDIF
 .ENDIF 
 .ENDIF 
 
--- scp2/source/ooo/ure.scp	18 Mar 2008 13:35:26 -0000	1.2
+++ scp2/source/ooo/ure.scp	21 Mar 2008 09:26:48 -0000
@@ -728,7 +730,7 @@
 End
 #endif
 
-#if defined _gcc3 && !(defined FREEBSD || defined NETBSD \
+#if defined _gcc3 && !(defined NETBSD \
 	|| defined MACOSX || defined SYSTEM_STDLIBS || defined(WNT))
 File gid_File_Dl_GccS
     TXT_FILE_BODY;
@@ -739,7 +741,7 @@
 End
 #endif
 
-#if defined _gcc3 && !(defined FREEBSD || defined NETBSD \
+#if defined _gcc3 && !(defined NETBSD \
 	|| defined MACOSX || defined SYSTEM_STDLIBS || defined(WNT))
 File gid_File_Dl_Stdcpp
     Dir = SCP2_URE_DL_DIR;

